feat(client): Add news translation

naomi-lgbt opened the pull request:
Signed-off-by: nhcarrigan <nhcarrigan@gmail.com>

Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [ ] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [ ] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [ ] My pull request targets the `master` branch of freeCodeCamp.
- [ ] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #XXXXX

<!-- Feel free to add any additional description of changes below this line -->

Lays the groundwork for allowing translations of the trending article links. With this foundation, it would now be possible to showcase the trending English articles for the English page, the trending Spanish articles for the Spanish page, etc.

Commit:
feat: Add translation for articles

Lays the groundwork for allowing translations of
the trending article links. With this foundation,
it would now be possible to showcase the trending
English articles for the English page, the trending
Spanish articles for the Spanish page, etc.

Signed-off-by: nhcarrigan <nhcarrigan@gmail.com>

nhcarrigan commented:
I still need to build/expand the tests for this addition - I'm stepping out for an appointment but will work on those when I return. 🙂

Commit:
tests: Incorporate `trending.json` into test

Includes the `trending.json` file in the expected
required files for the `locales` test suite. Also
adds a `trending-schema` to validate the data
structure is correct.

Signed-off-by: nhcarrigan <nhcarrigan@gmail.com>

Commit:
tests: Modify translation process

The tests on the footer errored out if the client
was last built in a language other than English.
This was due to the values showing up as hard-
coded in the jest snapshot. I've added the trending
json files to the i18n namespace list, allowing us
to parse them through the translation function. This
means they render in the snapshot as the variable
string, and the tests pass correctly. I additionally
fixed a typo in the class name.

Signed-off-by: nhcarrigan <nhcarrigan@gmail.com>
